### ISSUE TITLE
`define:vars` not working with `is:inline` directive in `<style>` tags

### ISSUE THREAD

**Issue by jacobdalamb** [withastro/compiler#1000]
### Astro Info

```block
Astro                    v4.5.9
Node                     v18.18.0
System                   Linux (x64)
Package Manager          npm
Output                   static
Adapter                  none
Integrations             none
```

### Describe the Bug

According to the Astro documentation, the `define:vars` directive should allow passing server-side variables from the component frontmatter into the `<style>` tags. The documentation does not mention any specific limitations or incompatibilities between `define:vars` and `is:inline` for <style> tags.

### What's the expected result?

When using the `define:vars` directive in a `<style>` tag that also has the `is:inline` directive, the variables passed through `define:vars` do not seem to be properly injected or accessible within the styles. The styles do not apply as expected, and there is an error message but goes away once the page is reloaded.

### Link to Minimal Reproducible Example

https://stackblitz.com/edit/github-zbqqwf?file=src%2Fpages%2Findex.astro

### Participation

- [ ] I am willing to submit a pull request for this issue.

**Comment by lilnasy** [comment]
For this input:
```astro
<style is:inline define:vars={{ textColor: foregroundColor, backgroundColor }}>
  h1 {
    background-color: var(--backgroundColor);
    color: var(--textColor);
  }
</style>
```
The compiler outputs:
```ts
$$render`<style>${$$defineStyleVars({ textColor: foregroundColor, backgroundColor })}
  h1 {
    background-color: var(--backgroundColor);
    color: var(--textColor);
  }
</style>`
```
...which renders a non-parseable inline stylesheet.

It should output:
```ts
$$render`<style${$$addAttribute({ textColor: foregroundColor, backgroundColor }, "define:vars")}>
  h1 {
    background-color: var(--backgroundColor);
    color: var(--textColor);
  }
</style>`
```

**Comment by Princesseuh** [comment]
Hello, the Go compiler for Astro is no longer maintained. All future development will be focused on [our new Rust compiler](https://github.com/withastro/compiler-rs). If this issue is still relevant and you can reproduce it on the Rust compiler, please open a new issue there. Thank you!
